cloud/clouds.go

```diff
@@ -281,6 +282,16 @@ func ParseCloudMetadata(data []byte) (map[string]Cloud, error) {
 		return nil, errors.Annotate(err, "cannot unmarshal yaml cloud metadata")
 	}
 
+	if err := validateCloudSet(data); err != nil {
```

Comment:
If we export the validate function, this can then be called from the cmd/juju/ package and it would output the warnings, so as to keep the user facing output close to the command code.